lib/abilities/template_conditions.rb

```diff
@@ -32,30 +39,31 @@ def partnership_templates(request_context)
 
       partnership_ids = Partnership.where(external_partnership_id: accessible_partnership_ids).pluck(:id)
 
-      Template.where(partnership_id: partnership_ids)
+      # Add global partnership if configured
+      partnership_ids << ExportLocation.global_partnership_id if ExportLocation.global_partnership_id.present?
+
+      Template.where(partnership_id: partnership_ids.uniq)
     end
 
     def entity(template, user:, ability: nil, request_context: nil)
       return true if template.account_id.blank? && template.partnership_id.blank?
 
       # Check request context first (from API params)
+      # If the template is a partnership template, we need to check the partnership context
       if request_context && request_context[:accessible_partnership_ids].present?
         return authorize_via_partnership_context(template, request_context)
       end
 
-      # Handle partnership templates - users don't have stored relationships anymore
-      # This should not be reached for partnership users since they use API context
-
-      # Handle regular account templates
-      return true if template.account_id == user.account_id
-      return false unless user.account&.linked_account_account
-      return false if template.template_sharings.to_a.blank?
-
-      account_ids = [user.account_id, TemplateSharing::ALL_ID]
+      # Handle partnership templates for account users (no API context)
+      if template.partnership_id.present?
+        return true if global_partnership_template?(template)
+        return true if user.access_token.present?
```

Comment:
Does this give access to partnership templates to any user with access tokens? I assume it's just for the cloning aspect.